packages/gatsby-source-shopify/src/create-operations.ts

```diff
@@ -68,6 +72,19 @@ export function createOperations(
         while (!finishedStatuses.includes(currentBulkOperation.status)) {
           await new Promise(resolve => setTimeout(resolve, 1000))
           currentBulkOperation = (await currentOperation()).currentBulkOperation
+
+          // add warning for CI environments
+          if (
+            process.env.CI &&
+            Date.now() > queryStartTime + (warningCount + 1) * warningInterval
+          ) {
+            warningCount += 1
+            const runtime = (warningInterval * warningCount) / 1000
```

Comment:
This time wont be accurate because it's 1 second + latency from currentOperation(), it's probably better to store a start time before the loop and compare the current time to that

Reply:
Updated!